common/headers/versionChecker.go

```diff
@@ -124,3 +131,24 @@ func (vc *versionChecker) ClientSupported(ctx context.Context, enableClientVersi
 
 	return nil
 }
+
+// ClientSupportsFeature returns true if the client reports support for the
+// given feature (which should be one of the Feature... constants above).
+func (vc *versionChecker) ClientSupportsFeature(ctx context.Context, feature string) bool {
+	headers := GetValues(ctx, SupportedFeaturesHeaderName)
+	clientFeatures := strings.Split(headers[0], SupportedFeaturesHeaderDelim)
+	for _, clientFeature := range clientFeatures {
+		if clientFeature == feature {
```

Comment:
Case insensitive comparison would be a nice convenience

Sending this feature list with every RPC call seems a bit wasteful on the wire compared to the config-file-based version mapping too. It's a relatively small but definitely nonzero bit of overhead

Reply:
Hmm, these are going to be hard-coded string constants.. what would case insensitivity buy us?

Sorry for the lack of context, we had a discussion about this among the server team: the plan is when the list gets long, group a bunch of them into super-features to reduce the length. Or eventually, stop supporting older versions and then we can remove some features and assume they're universal.

Reply:
It doesn't really buy you anything I suppose, no biggie. OK that sounds fine to me, we can always expand on things later.